Integrations/VulnDB/VulnDB.py

```
@@ -10,57 +10,42 @@
-                           headers={'Authorization': f'Bearer {get_oath_token()}'},
-                           params=params)
-        res.raise_for_status()
-        response = res.json()
-    except requests.exceptions.RequestException as err:
-        return_error(str(err))
-    except json.decoder.JSONDecodeError:
-        return_error("Failed to parse HTTP response to JSON. Original response: \n\n{}".format(res.text))
-    return response
+class Client(BaseClient):
+
+    def __init__(self, proxy, use_ssl, base_url, client_id, client_secret):
+        super().__init__(base_url=base_url, verify=use_ssl, proxy=proxy)
+        access_token = self.get_oath_token(client_id, client_secret)
+        headers = {'Content-Type': 'application/json',
+                   'Authorization': f'Bearer {access_token}'}
+        self._headers = headers
+
+    def get_oath_token(self, client_id, client_secret):
+        # Workaround ParseResult immutability
+        parse_result = list(urllib.parse.urlparse(self._base_url))
+        parse_result[2] = '/oauth/token'
+        oath_url = urllib.parse.urlunparse(parse_result)
+        res = self._http_request('POST',
+                                 '',
+                                 json_data={
+                                     'client_id': client_id,
+                                     'client_secret': client_secret,
+                                     'grant_type': 'client_credentials'
+                                 },
+                                 full_url=oath_url)
+        return res.get('access_token')
+
+    def http_request(self, url_suffix, size=None):
+        params = {'size': size} if size else None
+        res = self._http_request("GET",
+                                 url_suffix=url_suffix,
+                                 params=params)
+        for error_attribute in ['error', 'details']:
+            if error_attribute in res:
+                return_error(res[error_attribute])
```

raise exception instead of `return_error`